drivers/sensor/tdk/icp201xx/icp201xx_drv.c

```diff
@@ -0,0 +1,445 @@
+
+	return cfg->bus_io->write(&cfg->bus, reg, (uint8_t *)wbuffer, wlen);
+}
+
+#ifdef CONFIG_FPU
+static float convertToHeight(float pressure_kp, float temperature_C)
+{
+	return PRESSURE_TO_HEIGHT_COEFF * TO_KELVIN(temperature_C) *
+	       (LOG_ATMOSPHERICAL_PRESSURE - logf(pressure_kp));
+}
+#endif
+
+/*
+ * ICP201xx warm up.
+ * If FIR filter is enabled, it will cause a settling effect on the first 14 pressure values.
+ * Therefore the first 14 pressure output values are discarded.
+ *
+ */
+void inv_icp201xx_app_warmup(inv_icp201xx_t *icp_device, icp201xx_op_mode_t op_mode,
+			     icp201xx_meas_mode_t meas_mode)
+{
+	volatile uint8_t fifo_packets = 0;
+	uint8_t fifo_packets_to_skip = 14;
+
+	do {
+		fifo_packets = 0;
+		if (!inv_icp201xx_get_fifo_count(icp_device, (uint8_t *)&fifo_packets) &&
+		    (fifo_packets >= fifo_packets_to_skip)) {
+			uint8_t i_status = 0;
+
+			inv_icp201xx_flush_fifo(icp_device);
+
+			inv_icp201xx_get_int_status(icp_device, &i_status);
+			if (i_status) {
+				inv_icp201xx_clear_int_status(icp_device, i_status);
+			}
+			break;
+		}
+		inv_icp201xx_sleep_us(2000);
+	} while (1);
+}
+
+static int icp201xx_attr_set(const struct device *dev, enum sensor_channel chan,
+			     enum sensor_attribute attr, const struct sensor_value *val)
+{
+	int err = 0;
+	struct icp201xx_data *data = (struct icp201xx_data *)dev->data;
+
+	__ASSERT_NO_MSG(val != NULL);
+	icp201xx_mutex_lock(dev);
+	if (chan == SENSOR_CHAN_PRESS) {
+		if (attr == SENSOR_ATTR_CONFIGURATION) {
+			if ((val->val1 >= ICP201XX_OP_MODE0) && (val->val1 <= ICP201XX_OP_MODE4)) {
+				data->op_mode = val->val1;
+				err = inv_icp201xx_soft_reset(&(data->icp_device));
+				err = inv_icp201xx_config(&(data->icp_device), data->op_mode,
+							  ICP201XX_FIFO_READOUT_MODE_PRES_TEMP);
+				inv_icp201xx_app_warmup(&(data->icp_device), data->op_mode,
+							ICP201XX_MEAS_MODE_CONTINUOUS);
+			} else {
+				LOG_ERR("Not supported ATTR value");
+				icp201xx_mutex_unlock(dev);
+				return -EINVAL;
+			}
+#ifdef CONFIG_ICP201XX_TRIGGER
+		} else if (attr == SENSOR_ATTR_SLOPE_TH) {
+			data->pressure_change = *val;
+		} else if ((attr == SENSOR_ATTR_LOWER_THRESH) ||
+			   (attr == SENSOR_ATTR_UPPER_THRESH)) {
+			if (val->val1 > 0) {
+				data->pressure_threshold = *val;
+			} else {
+				LOG_ERR("Not supported ATTR value");
+				icp201xx_mutex_unlock(dev);
+				return -EINVAL;
+			}
+#endif
+		} else {
+			LOG_ERR("Not supported ATTR");
+			icp201xx_mutex_unlock(dev);
+			return -EINVAL;
+		}
+	};
+	icp201xx_mutex_unlock(dev);
+	return err;
+}
+
+static int icp201xx_sample_fetch(const struct device *dev, const enum sensor_channel chan)
+{
+	struct icp201xx_data *data = (struct icp201xx_data *)dev->data;
+	uint8_t fifo_packets;
+	uint8_t fifo_data[6];
+
+	icp201xx_mutex_lock(dev);
+
+	/** Read measurements count in FIFO **/
+	if (inv_icp201xx_get_fifo_count(&(data->icp_device), &fifo_packets)) {
```

Comment:
Nit: For Fetch/Get, Instead of servicing the FIFO: Why not just get the latest readings?

To consider for Sensor Model v2: If this were Read/Decode, structuring a single-shot reading like this would break an on-going data-stream.

Reply:
Good catch, no need to read the FIFO if you're going to throw away all but the latest.